src/grpc.cc

```diff
@@ -254,168 +258,124 @@ static grpc::Status unmarshal_value_list(const collectd::types::ValueList &msg,
 } /* unmarshal_value_list() */
 
 /*
- * request call-backs and call objects
+ * Collectd service
  */
-
-static grpc::Status Process(grpc::ServerContext *ctx,
-		DispatchValuesRequest request, DispatchValuesReply *reply)
-{
-	value_list_t vl = VALUE_LIST_INIT;
-	auto status = unmarshal_value_list(request.values(), &vl);
-	if (!status.ok())
-		return status;
-
-	if (plugin_dispatch_values(&vl))
-		status = grpc::Status(grpc::StatusCode::INTERNAL,
-				grpc::string("failed to enqueue values for writing"));
-	return status;
-} /* Process(): DispatchValues */
-
-static grpc::Status Process(grpc::ServerContext *ctx,
-		QueryValuesRequest request, QueryValuesReply *reply)
-{
-	uc_iter_t *iter;
-	char *name = NULL;
-
-	value_list_t matcher;
-	auto status = unmarshal_ident(request.identifier(), &matcher, false);
-	if (!status.ok())
-		return status;
-
-	if ((iter = uc_get_iterator()) == NULL) {
-		return grpc::Status(grpc::StatusCode::INTERNAL,
-				grpc::string("failed to query values: cannot create iterator"));
-	}
-
-	status = grpc::Status::OK;
-	while (uc_iterator_next(iter, &name) == 0) {
-		value_list_t res;
-		if (parse_identifier_vl(name, &res) != 0) {
-			status = grpc::Status(grpc::StatusCode::INTERNAL,
-					grpc::string("failed to parse identifier"));
-			break;
+class CollectdImpl : public collectd::Collectd::Service {
+public:
+	grpc::Status QueryValues(grpc::ServerContext *ctx, QueryValuesRequest const *req, grpc::ServerWriter<QueryValuesResponse> *writer) override {
+		value_list_t match;
+		auto err = unmarshal_ident(req->identifier(), &match, false);
```

Comment:
Nit: `status` and `err` are now used inconsistently in this file. `status` makes more sense to me b/c the value is not an error indicator (unlike, e.g., error variables in Go which are non-nil on error).